Sources/MountebankSwift/Api/Endpoint/Endpoint.swift

```diff
@@ -4,25 +4,44 @@ import Foundation
 /// [mbtest.org/docs/api/overview](https://www.mbtest.org/docs/api/overview)
 struct Endpoint {
     let method: HTTPMethod
-    let templatePath: String
+
+    private let templatePath: String
+    private let parameters: EndpointParameters?
+
+    private init(method: HTTPMethod, templatePath: String, parameters: EndpointParameters? = nil) {
+        self.method = method
+        self.templatePath = templatePath
+        self.parameters = parameters
+    }
 
     /// Get entry hypermedia
     // static func getHypermedia() = Endpoint(method: .get, templatePath: "/")
 
     /// Get a list of all imposters
-    static func getAllImposters() -> Endpoint { Endpoint(method: .get, templatePath: "/imposters") }
+    static func getAllImposters(parameters: ImposterParameters) -> Endpoint {
+        Endpoint(method: .get, templatePath: "/imposters", parameters: parameters)
+    }
 
     /// Get a single imposter
-    static func getImposter(port: Int) -> Endpoint { Endpoint(method: .get, templatePath: "/imposters/\(port)") }
+    static func getImposter(port: Int, parameters: ImposterParameters) -> Endpoint {
+        Endpoint(method: .get, templatePath: "/imposters/\(port)", parameters: parameters)
+    }
 
     /// Create a single imposter
-    static func postImposter() -> Endpoint { Endpoint(method: .post, templatePath: "/imposters") }
+    static func postImposter(parameters: ImposterParameters) -> Endpoint {
```

Comment:
Can you pass these query params when you're posting the imposters? And does that make sense?

Reply:
Good point, I've removed them.